### ISSUE TITLE
Bug: Can swap with zero amount

### ISSUE THREAD

**Issue by thien-do** [liquid-chain/explorer#15]
In the migrate tool, the user should not be able to swap with amount === 0

This is technically correct but in practice it does not make sense... the user would have nothing but still pay the fee

**Comment by sonhnguyenn** [comment]
do we need to validate this on backend as well

**Comment by thien-do** [comment]
@sonhnguyenn no, we only need it for the explorer here. Basically we'd need to set the Amount input to "number" type and have the "min" attribute

https://github.com/liquid-chain/explorer/blob/8aaabc31f32aaa7a31e4e72b86a04ccec3d1e616/components/migrate/connected/connected.tsx#L59

See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/min and https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number

**Comment by thien-do** [comment]
Hm I'm starting to think it's more complicated than it looks :)) The amountText is, well, a text so maybe a check in the submit function is easier. In other words, we can throw an error (and catch it and show it with Dialog.alert) here:

https://github.com/liquid-chain/explorer/blob/8aaabc31f32aaa7a31e4e72b86a04ccec3d1e616/components/migrate/connected/connected.tsx#L31-L35

**Comment by thien-do** [comment]
So the good UX would be the user should be able to enter whatever they want, even something like "1,000,000", then we will try to parse it to a number, and if the number is not valid we will tell them. This happens at the onSubmit only
